packages/cli/test/integration/workflows/workflows.controller.ee.test.ts

```diff
@@ -1236,3 +1242,308 @@ describe('PATCH /workflows/:workflowId - activate workflow', () => {
 		expect(active).toBe(false);
 	});
 });
+
+describe('PUT /:workflowId/transfer', () => {
+	test('cannot transfer into the same project', async () => {
+		const destinationProject = await createTeamProject('Team Project', member);
+
+		const workflow = await createWorkflow({}, destinationProject);
+
+		await testServer
+			.authAgentFor(member)
+			.put(`/workflows/${workflow.id}/transfer`)
+			.send({ destinationProjectId: destinationProject.id })
+			.expect(400);
+	});
+
+	test('cannot transfer into a personal project', async () => {
+		const destinationProject = await createTeamProject('Team Project', member);
+
+		const workflow = await createWorkflow({}, destinationProject);
+
+		await testServer
+			.authAgentFor(member)
+			.put(`/workflows/${workflow.id}/transfer`)
+			.send({ destinationProjectId: memberPersonalProject.id })
+			.expect(400);
+	});
+
+	test('cannot transfer without workflow:move scope for the workflow', async () => {
+		const destinationProject = await createTeamProject('Team Project', member);
+
+		const workflow = await createWorkflow({}, anotherMember);
+
+		await testServer
+			.authAgentFor(member)
+			.put(`/workflows/${workflow.id}/transfer`)
+			.send({ destinationProjectId: destinationProject.id })
+			.expect(403);
+	});
+
+	test('cannot transfer without workflow:create scope in destination project', async () => {
+		const destinationProject = await createTeamProject('Team Project', anotherMember);
+
+		const workflow = await createWorkflow({}, member);
+
+		await testServer
+			.authAgentFor(member)
+			.put(`/workflows/${workflow.id}/transfer`)
+			.send({ destinationProjectId: destinationProject.id })
+			.expect(404);
+	});
+
+	test('project members cannot transfer workflows', async () => {
```

Comment:
```suggestion
	test('user should not be able to transfer workflows to projects they are not part of', async () => {
```

Reply:
😬 I'll improve the name. This test tests that a `project:editor` cannot transfer workflows out of projects. That check happens before we check that the user has the `workflow:create` scope in the destination project so I didn't actually link the user to it. I'll amend that.